CONTRIBUTING.md

```diff
@@ -0,0 +1,99 @@
+# How to Contribute
+
+Flux is [Apache 2.0 licensed](LICENSE) and accepts contributions via GitHub
+pull requests. This document outlines some of the conventions on development
+workflow, commit message formatting, contact points and other resources to make
+it easier to get your contribution accepted.
+
+We gratefully welcome improvements to documentation as well as to code.
+
+# Certificate of Origin
+
+By contributing to this project you agree to the Developer Certificate of
+Origin (DCO). This document was created by the Linux Kernel community and is a
+simple statement that you, as a contributor, have the legal right to make the
+contribution. No action from you is required, but it's a good idea to see the
+[DCO](DCO) file for details before you start contributing code to Flux.
+
+# Chat
+
+The project uses Slack: To join the conversation, simply join the
+[Weave community](https://slack.weave.works/) Slack workspace and use the
+[#flux](https://weave-community.slack.com/messages/flux/) channel.
+
+## Getting Started
+
+- Fork the repository on GitHub
+- Read the [README](README.md) for getting started as a user and learn how/where to ask for help 
+- If you want to contribute as a developer, continue reading this document for further instructions
+- Play with the project, submit bugs, submit pull requests!
+
+## Contribution workflow
+
+This is a rough outline of how to prepare a contribution:
+
+- Create a topic branch from where you want to base your work (usually branched from master).
+- Make commits of logical units.
+- Make sure your commit messages are in the proper format (see below).
+- Push your changes to a topic branch in your fork of the repository.
+- If you changed code:
+   - add automated tests to cover your changes
+- Submit a pull request to the original repository.
+
+## How to build and run the project
+
+Refer to the [building doc](site/building.md) to find out how to build from
+source.
+
+## How to run the test suite
+
+You can run the linting and unit tests by simply doing
+
+```bash
+make test
+```
+
+# Acceptance policy
+
+These things will make a PR more likely to be accepted:
+
+ * a well-described requirement
+ * tests for new code
+ * tests for old code!
+ * new code and tests follow the conventions in old code and tests
+ * a good commit message (see below)
+
+In general, we will merge a PR once two maintainers have endorsed it.
+Trivial changes (e.g., corrections to spelling) may get waved through.
+For substantial changes, more people may become involved, and you might
+get asked to resubmit the PR or divide the changes into more than one PR.
+
+### Format of the Commit Message
+
+We follow a rough convention for commit messages that is designed to answer two
+questions: what changed and why. The subject line should feature the what and
+the body of the commit should describe the why.
+
+```
```

Comment:
This is what I prefer: https://chris.beams.io/posts/git-commit/

(In particular I don't like having `subsystem:` because it takes up valuable space that could be used for description).

I would sum it up, borrowing from https://chris.beams.io/posts/git-commit/#seven-rules, as

 1. Limit the subject to 50 characters and write as the continuation of the sentence "If applied, this commit will ..."
 2. Explain what and why in the body, if more than a trivial change; wrap it at 72 characters.
 3. (then read the linked article for more details!)